scripts/sysdig-deploy/chart-version-bump.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+set -e
+
+na_chart_path="charts/node-analyzer"
+agent_chart_path="charts/agent"
+kspm_collector_chart_path="charts/kspm-collector"
+rapid_response_chart_path="charts/rapid-response"
+sysdig_deploy_path="charts/sysdig-deploy"
+
+minor=0
+major=0
+patch=0
+
+check_upgrade_type () {
+    # possible value of $changes might be as follows:
+    # -version: 0.1.1
+    # +version: 0.1.4
+    before=$(echo "$changes" | grep "\-version:" | sed "s/-version:[[:space:]]*//")
+    after=$(echo "$changes" | grep "\+version:" | sed "s/+version:[[:space:]]*//")
+    IFS='.'
+
+    read -ra prev_ver <<< "$before"
+    read -ra next_ver <<< "$after"
+    
+    echo "prev: ${prev_ver[@]}"
+    echo "after: ${next_ver[@]}"
+    
+    for ((idx=0; idx<3; ++idx)); do
+        if [ ${next_ver[idx]} -gt ${prev_ver[idx]} ] 
+        then
+            if [ $idx -eq 0 ]
+            then
+                major=1
+            elif [ $idx -eq 1 ]
+            then 
+                minor=1
+            elif [ $idx -eq 2 ]
+            then
+                patch=1
+            fi
+        fi
+    done
+}
+
+update_sysdig_deploy_chart () {
+    curr_version=$(cat "$sysdig_deploy_path/Chart.yaml" | grep "version:" -m 1 | sed "s/version:[[:space:]]*//")
+    echo "sysdig-deploy version : $curr_version"
+    IFS='.'
+    read -ra sysdig_deploy_version <<< $curr_version
+
+    new_major=$((${sysdig_deploy_version[0]}))
+    new_minor=$((${sysdig_deploy_version[1]}))
+    new_patch=$((${sysdig_deploy_version[2]}))
+
+    if [ $major -eq 1 ]
+    then
+        new_major=$(($new_major + 1))
+        new_minor=0
+        new_patch=0
+    elif [ $minor -eq 1 ]
+    then
+        new_minor=$(($new_minor + 1))
+        new_patch=0
+    elif [ $patch -eq 1 ]
+    then
+        new_patch=$(($new_patch + 1))
+    fi
+
+    final_sysdig_deploy_version=$(echo "$new_major.$new_minor.$new_patch")
+    echo "Final deploy version is to be : $final_sysdig_deploy_version"
+    sed -i -e "1,/version: .*/ s/version: .*/version: ${final_sysdig_deploy_version}/" "$sysdig_deploy_path/Chart.yaml"
+}
+
+output=$(git diff --name-only HEAD~1 HEAD -- $na_chart_path $agent_chart_path)
```

Comment:
do i understand correctly that this will trigger the bump only if the version is changed in node-analyzer or agent? if yes is it correct?